### PR TITLE
update fix for GLFW error handler to 2.204

### PR DESCRIPTION
**Pull request by sse2**
instead of copying function over, patch call ds:MessageBoxW to call ds:MessageBoxA
this is better because now there's 2 different functions for GLFW error handling(???) and they have different calling conventions

before https://i.imgur.com/HCp7d1v.png
after https://i.imgur.com/zzPELWK.png

### CODE REVIEW

File: loader/src/hooks/MessageBoxFix.cpp [hunk 1]
```diff
@@ -28,11 +28,25 @@ static void __cdecl fixedErrorHandler(int code, char const* description) {
...
-    //    reinterpret_cast<void*>(geode::base::getCocos() + 0x19feec), toByteArray(&fixedErrorHandler)
-    //);
+$execute {
+    // updated for 2.204
+    // check xrefs to "GLFWError #%d Happen, %s\n", now there's two functions with the same exact behaviour, one is a member function though...
+	// call ds:MessageBoxW 
+    // patch MessageBoxW to MessageBoxA
+    // geode::base::getCocos() + 0x122600 = MessageBoxA in .idata
+    // geode::base::getCocos() + 0x1225DC = MessageBoxW in .idata
+    const uint32_t importedMessageBoxA = (geode::base::getCocos() + 0x122600);
```
Comment (matcool) [inline loader/src/hooks/MessageBoxFix.cpp:38]: why the ()

File: loader/src/hooks/MessageBoxFix.cpp [hunk 2]
```diff
@@ -28,11 +28,25 @@ static void __cdecl fixedErrorHandler(int code, char const* description) {
...
+    
+    ByteVector p = {
+        (importedMessageBoxA) & 0xff,
+        (importedMessageBoxA >> 8) & 0xff,
+        (importedMessageBoxA >> 16) & 0xff,
+        (importedMessageBoxA >> 24) & 0xff};
+
+    (void)Mod::get()->patch(reinterpret_cast<void*>(geode::base::getCocos() + 0xC75F9), p
+    );
```
Comment (matcool) [inline loader/src/hooks/MessageBoxFix.cpp:47]: why the ); on another line..

**Review by matcool — CHANGES_REQUESTED** [review]

**Comment by matcool** [comment]
if possible this should check for if the loader is in forwardCompatMode, except i think $execute is too early for that?

File: loader/src/hooks/MessageBoxFix.cpp (hunk 1 above)
Comment (sse2) [reply loader/src/hooks/MessageBoxFix.cpp:38]: forgot to remove because i casted before, fixing rn

File: loader/src/hooks/MessageBoxFix.cpp (hunk 2 above)
Comment (sse2) [reply loader/src/hooks/MessageBoxFix.cpp:47]: visual studio stinks im fixing rn

**Comment by sse2** [comment]
Added check for forwardcompatmode, works fine, checked by manually changing the timestamp of geometrydash.exe

**Pull request merged**